run.ts

```diff
@@ -63,37 +63,33 @@ const launchDocker: SearchEngineLauncherFunction = async ({
   port,
   options,
 }) => {
-  const Image =
-    engine === 'elasticsearch'
-      ? 'elastic/elasticsearch:8.6.2'
-      : 'opensearchproject/opensearch:2.11.0'
-  console.log('Launching Docker container', Image)
-  const docker = new Dockerode()
-  const container = await docker.createContainer({
-    Env: [...options, 'path.data=/var/lib/search', 'path.logs=/var/log/search'],
-    HostConfig: {
-      AutoRemove: true,
-      Mounts: [
-        { Source: dataDir, Target: '/var/lib/search', Type: 'bind' },
-        { Source: logsDir, Target: '/var/log/search', Type: 'bind' },
-      ],
-      PortBindings: {
-        [`${port}/tcp`]: [{ HostIP: '127.0.0.1', HostPort: `${port}` }],
-      },
-    },
-    Image,
+  const subprocess = fork(
+    './node_modules/@nasa-gcn/architect-plugin-search/launchSearch.js',
+    [dataDir, logsDir, engine, port, options] as ForkOptions
+  )
+
+  subprocess.stdout?.on('data', (data) => {
+    console.log(`subprocess stdout: ${data}`)
+  })
+
+  subprocess.stderr?.on('data', (data) => {
+    console.error(`subprocess stderr: ${data}`)
+  })
+  const waitUntilStopped = new Promise<void>((resolve) => {
+    subprocess.on('message', (message) => {
+      if (message === 'containerStopped') {
+        resolve()
+      }
+    })
   })
-  const stream = await container.attach({ stream: true, stderr: true })
-  stream.pipe(process.stderr)
-  await container.start()
 
   return {
     async kill() {
       console.log('Killing Docker container')
-      await container.kill()
+      subprocess.send({ action: 'kill' })
```

Comment:
`subprocess.send` is asynchronous and takes a callback. I think that you should `await` it.